Tutorials/Members-Registration-And-Logins/index.md

```diff
@@ -0,0 +1,245 @@
+                    throw new NotImplementedException(status.ToString());
+                case MembershipCreateStatus.InvalidEmail:
+                    ModelState.AddModelError("registerModel.Email", "Email is invalid");
+                    break;
+                case MembershipCreateStatus.DuplicateUserName:
+                    ModelState.AddModelError((model.UsernameIsEmail || model.Username == null)
+                        ? "registerModel.Email"
+                        : "registerModel.Username",
+                        "A member with this username already exists.");
+                    break;
+                case MembershipCreateStatus.DuplicateEmail:
+                    ModelState.AddModelError("registerModel.Email", "A member with this e-mail address already exists");
+                    break;
+                case MembershipCreateStatus.UserRejected:
+                case MembershipCreateStatus.InvalidProviderUserKey:
+                case MembershipCreateStatus.DuplicateProviderUserKey:
+                case MembershipCreateStatus.ProviderError:
+                    ModelState.AddModelError("registerModel", "An error occurred creating the member: " + status);
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
+
+            return CurrentUmbracoPage();
+        }
+
+        //the below method will allow us to assign the member to an already existing group
+        private void AssignMemberGroup(string email, string group)
+        {
+            try
+            {
+                Services.MemberService.AssignRole(email, group);
+            }
+            catch (Exception ex)
+            {
+                //handle the exception
+            }
+
+        }
+
+    }
+}
+```
+
+For an easier implementation, you can copy the above code to a new .cs file and place it in the App_Code folder in your solution so that it will be compiled on application startup.
+
+With the above controller in place it is time to adjust the macro/view files as well.
+
+## Adjusting the Registration partial to use the new controller
+
+In the Backoffice, navigate to the Register partial you created before. Where we would normally be using
+
+```csharp
+    using (Html.BeginUmbracoForm<UmbRegisterController>("HandleRegisterMember"))
+    {
+        //rest of the snippet code
+    }
+```
+
+we have to instead use the custom controller we added, as well as include an anti-forgery token:
+
+```csharp
+ using (Html.BeginUmbracoForm<UmbAlternativeRegisterController>("HandleRegisterMember", new { memberGroup = "Professionals" }))
+    {
+            @Html.AntiForgeryToken()
+        //rest of the snippet code
+    }
+```
+
+We are also passing a member group as a parameter - people who register with this form will be automatically assigned to the "Professionals" member group, assuming it already exists in the Backoffice.
```

Comment:
Doesn't this mean that a step here would be to create the new Member Group **before** setting up the controller itself? 🤔

Reply:
no, the member group is set up along the way around step (image) 12

Reply:
Since most of our tutorials are linear I assume people following the guide have already created a member group as that is explained earlier